src/utils/nameUtils.ts

```diff
@@ -4,7 +4,70 @@ import { Operation, OperationGroup } from "@azure-tools/codemodel";
 import { getLanguageMetadata } from "./languageHelpers";
 import { TypeDetails, PropertyKind } from "../models/modelDetails";
 
-const ReservedModelNames = ["Error", "Date"];
+const ReservedModelNames = [
```

Comment:
Should we be using one set of rules for all models?

I can see one set of rules for things that can be 'new'd up. For example, OperationGroups can be instantiated directly by the user, so you need to guard against names like `Date`, `String`, etc. 

However, properties can be pretty much anything. You can have an interface that looks like this:
```ts
interface Foo {
  string: string;
}
```
and that works just fine. Likewise for parameters:
```ts
function(string: string) {}
```
However, in the parameters case you definitely don't want to allow any reserved words (and that's probably true for properties too so that params/properties stay consistent across operation inputs and outputs.

Side-note: Can you make this list ordered alphabetically? It'll be easier to add things or check what's in it then.

Reply:
Alright, thanks @daviwil and @chradek for your feedback around this. I agree with you and here's what I'm thinking:

* Have metadata on the reserved words to indicate in which cases we want special handling. So that we just maintain a single list of reserved words but can choose case by case if we need to suffix the name.

* File, operations and properties, don't need suffixing as @chradek mentioned above, so I have skipped a suffix in those cases.

* OperationGroups, Parameters, Classes and Interfaces(for consistency) would need to be guarded against a subset of reserved words, such as Error, String, Date, Boolean, etc
   * OperationGroups will get Operations suffix.
   * Parameters would get Param suffix
   * Classes and Interfaces will get Model suffix